dialogflow/stopwatch/README.md

```diff
@@ -1,42 +1,72 @@
-- Be sure that you have gone through the steps in the [Dialogflow quickstart](https://cloud.google.com/dialogflow-enterprise/docs/quickstart) to create and configure your stopwatch agent. Helper scripts for this are in the `SETUP_SERVICE` directory.
-- Run ./INSTALL-COCOAPODS to install app dependencies. When it finishes, it will open the Stopwatch workspace in Xcode. Since we are using Cocoapods, be sure to open the workspace and not Stopwatch.xcodeproj.
-- Replace `your-project-identifier` in `StopwatchService.swift` with the identifier of your Google Cloud project.
+## Setup
+- Create a project with the [Google Cloud Console][cloud-console]
+- Enable the [Dialogflow API](https://console.cloud.google.com/apis/library/dialogflow.googleapis.com).
+- [Enable billing][billing].
+- [Import the Dialogflow Agent](https://dialogflow.com/docs/agents/export-import-restore#import) using the `StopwatchAgent.zip` which is located in the `stopwatch` directory. 
+- [Create a Service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts) with the following IAM role: `Dialogflow API Client`. Example name: `dialogflow-client`. ([For more info on: how to add roles to a Service Account](https://cloud.google.com/iam/docs/granting-roles-to-service-accounts#granting_access_to_a_service_account_for_a_resource))
+- For your "App Engine Default Service Account" add the following role `Service Account Token Creator` Role. ([For more info on: how to add roles to a Service Account](https://cloud.google.com/iam/docs/granting-roles-to-service-accounts#granting_access_to_a_service_account_for_a_resource))
+- To use Sentiment Analysis you need to [Enable beta features](https://cloud.google.com/dialogflow-enterprise/docs/sentiment#enable_beta_features) 
+- To use Text-to-Speech you need to [Enable beta features](https://cloud.google.com/dialogflow-enterprise/docs/detect-intent-tts#enable_beta_features)
+- To use Knowledge Connectors you need to [Enable beta features](https://cloud.google.com/dialogflow-enterprise/docs/knowledge-connectors#enable_beta_features) 
+
+
+### Setup the app
+- Clone this repository `git clone https://github.com/GoogleCloudPlatform/ios-docs-samples.git` 
+- `cd ios-docs-samples/dialogflow/stopwatch/` 
+- Run `./INSTALL-COCOAPODS` to install app dependencies (this can take a long time to run). When it finishes, it will open the Stopwatch workspace in Xcode. Since we are using Cocoapods, be sure to open the `Stopwatch.xcworkspace` and not `Stopwatch.xcodeproj`.
+- Replace `your-project-identifier` in `ApplicationConstants.swift` with the identifier of your Google Cloud project.
+
+###  Setup Firebase on the application:
+
+- Complete the steps for [Add Firebase to your app](https://firebase.google.com/docs/ios/setup#add_firebase_to_your_app). Note: No need to complete any other sections, they are already done. 
```

Comment:
This doesn't tell me how to add Firebase to my app - the linked instructions say "Click Add Firebase to your iOS app" but not where I need to go to find that link.

Reply:
In the linked instructions screen, Please expand the "Create a Firebase project" section to find the detailed steps to add a project to Firebase console. I have added this content as part of README.md.